torch/autograd/grad_mode.py

```diff
@@ -24,6 +24,9 @@ def __call__(self, func: F) -> F:
         def decorate_context(*args, **kwargs):
             with self:
                 return func(*args, **kwargs)
+
+        # adding this attribute to check if this is called from JIT
+        decorate_context.__gradDecorator = True
```

Comment:
Why don't you just set that unconditionally at the beginning of the call function?
Also we could make this a non-private name that explains what it is doing and what this can be used used for?

Reply:
Hi Alban,
Presently I am closing this pull request; as this functionality os not supported by PyTorch as off now; will analyze further and see if the same can be solved by some other means.